core/block_validator.go

```diff
@@ -94,9 +94,23 @@ func (v *BlockValidator) ValidateBody(block *types.WorkObject) error {
 		if hash := types.CalcUncleHash(block.Uncles()); hash != header.UncleHash() {
 			return fmt.Errorf("uncle root hash mismatch: have %x, want %x", hash, header.UncleHash())
 		}
-		if v.hc.ProcessingState() {
-			if hash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil)); hash != header.TxHash() {
-				return fmt.Errorf("transaction root hash mismatch: have %x, want %x", hash, header.TxHash())
+		if hash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil)); hash != header.TxHash() {
+			return fmt.Errorf("transaction root hash mismatch: have %x, want %x", hash, header.TxHash())
+		}
+		activeLocations := common.NewChainsAdded(v.hc.currentExpansionNumber)
+		for _, tx := range block.Transactions() {
+			if types.QiTxType == tx.Type() {
+				for _, txo := range tx.TxOut() {
+					found := false
+					for _, activeLoc := range activeLocations {
+						if common.IsInChainScope(txo.Address, activeLoc) {
+							found = true
+						}
+					}
+					if !found {
```

Comment:
wouldnt be easier to check if its not in scope and returning instead of going through all outputs even if the first outpoint does not follow this rule? I dont mind this, just curious

Reply:
Its ok to emit out of scope UTXOs... just not to a scope which isn't active yet

Reply:
never mind, it looks good, idk why it was confusing